cpp/src/dictionary/dictionary_column_view.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/dictionary/dictionary_column_view.hpp>
+#include <cudf/utilities/error.hpp>
+
+namespace cudf
+{
+
+//
+dictionary_column_view::dictionary_column_view( column_view const& strings_column )
+    : column_view(strings_column)
+{
+    CUDF_EXPECTS( type().id()==DICTIONARY32, "dictionary_column_view only supports DICTIONARY type");
+}
+
+column_view dictionary_column_view::parent() const
+{
+    return static_cast<column_view>(*this);
+}
+
+column_view dictionary_column_view::indices() const
+{
+    CUDF_EXPECTS( num_children()>0, "dictionary column has no children" );
```

Comment:
No. An empty dictionary will not have any children so we want the ctor to succeed whether there is children or not. This is the same as for `strings_column_view` .

Reply:
Right, I was thinking the check would look something like:
```c++
if(size > 0){
   CUDF_EXPECTS(num_childre()==2);
}
```
That way we communicate that there is an error at construction rather than later on in a member function.